services/pingpong/pingpong-status.service.js

```diff
@@ -0,0 +1,68 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService, InvalidParameter } = require('..')
+
+const schema = Joi.object({
+  status: Joi.string().required(),
+}).required()
+
+const pingpongDocumentation = `
+<p>
+  To see more details about this badge and obtain your api key, visit
+  <a href="https://my.pingpong.one/integrations/badge-status/" target="_blank">https://my.pingpong.one/integrations/badge-status/</a>
+</p>
+`
+
+module.exports = class PingPongStatus extends BaseJsonService {
+  static category = 'monitoring'
+  static route = { base: 'pingpong/status', pattern: ':apiKey' }
+
+  static examples = [
+    {
+      title: 'PingPong status',
+      namedParams: { apiKey: 'sp_2e80bc00b6054faeb2b87e2464be337e' },
+      staticPreview: this.render({ status: 'Operational' }),
+      documentation: pingpongDocumentation,
+      keywords: ['statuspage', 'status page'],
+    },
+  ]
+
+  static defaultBadgeData = { label: 'status' }
+
+  static validateApiKey({ apiKey }) {
+    if (!apiKey.startsWith('sp_')) {
+      throw new InvalidParameter({
+        prettyMessage: 'invalid api key',
+      })
+    }
+  }
+
+  static render({ status }) {
+    switch (status) {
+      case 'Operational':
+        return { message: 'up', color: 'brightgreen' }
+      case 'Major issues':
+        return { message: 'issues', color: 'orange' }
+      case 'Critical state':
+        return { message: 'down', color: 'red' }
+      case 'Maintenance mode':
+        return { message: 'maintenance', color: 'lightgrey' }
+      default:
+        return { message: 'unknown', color: 'lightgrey' }
```

Comment:
This pairs with my comment above on the schema

Any occurrence that hits this branch would mean that something's changed upstream that this service class in Shields isn't prepared to handle. That's an error scenario that we want to acknowledge loudly, whereas returning a non-error badge with an `unknown` message could give the user the incorrect impression that PingPong doesn't know the status of their site